tests/templates/test-templates.php

```diff
@@ -75,28 +76,141 @@ public function test_template_paths( $paths ) {
 	 * Test that local template partial paths is working.
 	 */
 	public function test_template_partial_paths() {
-		$path = Templates\locate_template_part( 'sidebar' );
+		$path = locate_template_part( 'sidebar' );
 
 		$this->assertTrue( file_exists( $path ) );
 	}
 
 	/**
-	 * Test default template context.
+	 * Test data hydration.
 	 *
-	 * @group context
+	 * @dataProvider get_template_test_cases
+	 * @group hydration
+	 *
+	 * @param array  $data     Test data to hydrate.
+	 * @param array  $expected Expected result from hydration.
+	 * @param string $message  Optional. Failure message.
 	 */
-	public function test_template_default_context() {
-		// Override the global post object for this test.
-		global $post;
+	public function test_hydrate_template( $data, $expected, $message = '' ) {
+		$this->assertEquals( $expected, hydrate_template( $data ), $message );
+	}
 
-		// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
-		$post = $this->factory()->post->create_and_get();
+	/**
+	 * Data provider for template hydration tests.
+	 *
+	 * @return array[] List of params.
+	 */
+	public function get_template_test_cases() {
+		return [
+			// Basic hydration.
+			[
+				// Template.
+				[
+					[ 'name' => 'test/component' ],
+					[ 'name' => 'test/component' ],
+				],
+				// Expected.
+				[
+					new Component( 'test/component' ),
+					new Component( 'test/component' ),
+				],
+				'Could not confirm template hydration.',
+			],
+			// Basic parent child.
+			[
+				// Template.
+				[
+					[
+						'name'     => 'test/parent',
+						'config'   => [],
+						'children' => [
+							[
+								'name'   => 'test/child',
+								'config' => [],
+							],
+						],
+					],
+				],
+				// Expected.
+				[
+					new Component(
+						'test/parent',
+						[
+							'children' => [
+								'test/child',
+							],
+						]
+					),
+				],
+				'Could not confirm parent/child hydration.',
+			],
+			// Basic template part hydration.
+			// Loaded from the template-parts/example.json.
+			[
+				// Template.
+				[
+					[ 'name' => 'template-part/example' ],
+				],
+				// Expected.
+				[
+					new Component( 'example/component1' ),
+					new Component(
+						'example/component2',
+						[
+							'children' => [
+								'example/component3',
+							],
+						]
+					),
+				],
+				'Template parts are not hydrating correctly.',
+			],
+			// Nested template part hydration.
+			// Loaded from the template-parts/example-nested.json.
+			[
+				// Template.
+				[
+					[ 'name' => 'template-part/example-nested' ],
+				],
+				// Expected.
+				[
+					new Component( 'example/component' ),
+					new Component( 'example/component1' ),
+					new Component(
+						'example/component2',
+						[
+							'children' => [
+								'example/component3',
+							],
+						]
+					),
+					new Component( 'example/component' ),
+				],
+				'Nested template parts are not hydrating correctly.',
+			],
```

Comment:
Are there other specific test cases that we want to add here?

Reply:
Nested template parts?

Reply:
Are you asking for an explanation of what a nested template part is? I was thinking of a case where you might have a template partial that references another template partial. We should support this use case and wanted to write a test that explicitly confirms that it's working.